### PR TITLE
Create main.js

### PR DESCRIPTION
**Pull request by t-amberwang**

### CODE REVIEW

File: lib/main.js [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+const ERROR_THRESHOLD = parseFloat(core.getInput('errorThreshold')) // max error tolerance on http pings
+const TIME_CONVERSION = 60000 // conversion factor from minutes to milliseconds
+
+let RESOURCE = ""
+let prevTrafficSettings = ""
+
+// utility function for async exec command
+const execute = function (command) {
+  return new Promise((resolve, reject) => {
+    console.log(`Running command ${command} at ${getDateTime()}`)
```
Comment (shekhar-gupta-dev) [inline lib/main.js]: Good to create a separate logging class and method something like 

class Logger {
 static LogInfo(message) { ... }
 static LogError(exception) { ... }
  static #privateLogInternal(params) { ... }
}

These methods can call an common Private method to log the events in a more uniform structure like:
2022-28-07 11:47Z      INFO     Running command foobar ...
2022-28-07 11:48Z      INFO     Some more log message here.
2022-28-07 11:48Z      ERROR  Exception encountered - some exception here.


Now, you can use this logger in your main code like this:
<pre>
Logger.LogInfo(`running command ${cmd}`);
Logger.LogError(`{error}`);
</pre>

File: lib/main.js [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+  })
+}
+
+// sleep callback with delay in ms
+const sleep = (delay) => new Promise(r => setTimeout(r, delay));
+
+// get date and time for queries
+function getDateTime() {
+  var today = new Date();
+  var date = today.getFullYear() + '-' + (today.getMonth() + 1) + '-' + today.getDate();
```
Comment (shekhar-gupta-dev) [inline lib/main.js]: you can just do this as well:
date = today.toISOString().split('T')[0]

File: lib/main.js [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+  var time = today.getHours() + ":" + today.getMinutes() + ":" + today.getSeconds();
+  return date + 'T' + time;
```
Comment (shekhar-gupta-dev) [inline lib/main.js]: Infact, you can get YYYY-MM-DDThh:mm:ss.sssZ format by simply doing 
today.toISOString()

File: lib/main.js [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+    await sleep(MONITOR_INTERVAL * TIME_CONVERSION)
+    console.log(`Running tests...`)
+    await runTests(startTime)
+  }
+}
+
+// function to run tests
+async function runTests(startTime) {
+  // run user provided tests by pinging apis
+  if (MONITORS) {
```
Comment (shekhar-gupta-dev) [inline lib/main.js]: good to use more meaningful variable names for better readability of code.
I am guessing Monitors are api endpoints to hit (based on the comments at declaration)
If so, consider using the name 'apiEndpointsToTest'

File: lib/main.js [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+    for (const test of MONITORS) {
+      core.debug("testing user provided monitors")
+      try {
+        let response = await fetch(test)
+        if (response.ok) {
```
Comment (shekhar-gupta-dev) [inline lib/main.js]: consider [200-299] range as the success.

**Comment by ShekharGupta1988** [comment]
hey @t-amberwang, 
Lets discuss the code structuring a bit today and break this one monolithic file into separate components (something like below). It also separates the responsibilities for various components and easy to manage for future.

image

**Pull request merged**